addons/portal/models/portal_mixin.py

```diff
@@ -95,15 +95,17 @@ def get_access_action(self, access_uid=None):
                 record.check_access_rights('read')
```

Comment:
Isn't it strange that we check the access rights against the current user and not access_uid?
Anyway, not related to your changes I guess.

Reply:
Because if current user has no access to the record, he cannot try to call this method with access_uid=SUPERUSER and gain access to it through token or stuff like that. If he cannot read it, he cannot try to have other people's access details. Especially when this method was public.